internal/exec/engine.go

```diff
@@ -127,6 +127,13 @@ func (e Engine) Run(stageName string) error {
 		return err
 	}
 
+	// On AWS we want to retrieve the region hint from ec2metadata. The fetch stages
+	// perform this during the fetching of the config (to ensure that networking is
+	// already up).
+	if e.PlatformConfig.Name() == "aws" && !strings.HasPrefix(stageName, "fetch") {
```

Comment:
I'm not thrilled by this special case.  Would it make sense to add an initialization function in `struct Config` in `platform.go`, which is called in all stages (and in the fetch stages, is called before the provider config is fetched)?  That would keep the region hint setup local to `aws.go`.

Reply:
> (and in the fetch stages, is called before the provider config is fetched)

So the reason I drifted more to this path was that this was initially broken in https://github.com/coreos/ignition/commit/877735ca4210ac2f7fbb1dca96378dbbfe3141c4 when we moved the fetching of the `regionHint` to after the config fetch to prevent racing the metadata server.

Maybe instead of having it in the initialization function we create a generic `PostSetup` (pls bikeshed the name if you like the idea) which we run on all stages post config fetch which happens to do the fetch of the `RegionHint` on `aws`?

Alternatively I can add a retry of the `RegionHint` fetch and move it back into the `NewFetcher` function pre-config fetch.

Reply:
Ah, yeah, okay.  I'm okay just defining the initialization function to run after config fetch.